spec/controllers/groups_controller_spec.rb

```diff
@@ -135,8 +135,38 @@
     end
 
     describe '#rename_group'
-    describe '#valid_grouping'
-    describe '#invalid_grouping'
+
+    describe '#valid_grouping' do
+      let(:grouping) { create :grouping_with_inviter }
+      it 'should validate a previously invalid grouping' do
+        post_as instructor, :global_actions, params: { course_id: course.id,
+                                                       assignment_id: grouping.assignment.id,
+                                                       groupings: [grouping.id],
+                                                       global_actions: 'valid' }
+
+        expect(grouping.reload.instructor_approved).to be true
+      end
+    end
+
+    describe '#invalid_grouping' do
+      let(:grouping) { create :grouping_with_inviter, instructor_approved: true }
+      let(:grouping2) { create :grouping_with_inviter, instructor_approved: false }
```

Comment:
Should I be doing the same with grouping too? Since grouping would also only apply for just one test. Or should I used let! and leave grouping as is for future possible tests.

Reply:
@therealfake good question, I think there was a misunderstanding with the purpose of the test. Your previous version passed `grouping.id` to the controller method, not `grouping2.id`, so I thought the test was checking to make sure that `grouping2` was unchanged.

But, the way you've changed the tests, you're passing `grouping2.id` instead. This is also a good test, and I don't care too much about the previous one. So in this case, what you should do is keep what you had before (both `grouping` and `grouping2` defined as `let`s), *but* pick more meaningful names like `valid_grouping` and `invalid_grouping`, so that it's more obvious what each test is doing. (And please make the same change for both `valid_grouping` and `invalid_grouping`.)

Reply:
I think the misunderstanding was due to my leaving out ```grouping.``` in front of ```course.id``` for the ```course_id``` parameter in a previous change. I have renamed the objects to be more meaningful and didn't use the names valid_grouping/invalid_grouping because those refer to the route.